src/compile/facet.ts

```diff
@@ -108,14 +108,14 @@ export class FacetModel extends Model {
     }, {});
   }
 
-  private _initAxis(facet: Facet, config: Config, child: Model): Dict<Axis> {
+  private _initAxis(facet: Facet, child: Model): Dict<Axis> {
     const model = this;
     return [ROW, COLUMN].reduce(function(_axis, channel) {
       if (facet[channel]) {
         const axisSpec = facet[channel].axis;
         if (axisSpec !== false) {
           const modelAxis = _axis[channel] = extend({},
-            config.facet.axis,
+            defaultFacetAxisConfig,
```

Comment:
Similar to [`_initAxis` for `UnitModel`](https://github.com/vega/vega-lite/pull/1944/files#r100684723), I don't think this is correct. 

However, just simply removing it will make things wrong too.  
Because facet's axis should *always* hide `domain`, `grid` and `ticks` but currently we do so by relying on `defaultFacetAxisConfig`, which is still configurable.

We should remove these 3 properties from `defaultFacetAxisConfig` (Notice [TODO](https://github.com/vega/vega-lite/pull/1944/commits/aca82ca8c8ba034ee4b1d03712e66b524c1eed9a#diff-a8e48aa61f457230707f1f7e707e76e5R230) in their declaration?) and instead add rules for  `domain`, `grid` and `ticks` to always return `false` for facet (`row` and `column` channel)
(For easy review, do this removal + add test in a separate commit)

https://github.com/vega/vega-lite/pull/1944/files#diff-a8e48aa61f457230707f1f7e707e76e5R109